api/src/main/java/org/apache/cloudstack/api/BaseCmd.java

```diff
@@ -247,30 +247,36 @@ public void setResponseType(final String responseType) {
         this.responseType = responseType;
     }
 
-    /**
-     * For some reason this method does not return the actual command name, but more a name that
-     * is used to create the response. So you can expect for a XCmd a value like xcmdresponse. Anyways
-     * this methods is used in too many places so for now instead of changing it we just create another
-     * method {@link BaseCmd#getActualCommandName()} that returns the value from {@link APICommand#name()}
-     *
-     * @return
-     */
-    public abstract String getCommandName();
-
-
     /**
      * Gets the CommandName based on the class annotations: the value from {@link APICommand#name()}
      *
      * @return the value from {@link APICommand#name()}
      */
-    public String getActualCommandName() {
+    public static String getCommandNameByClass(Class<?> clazz) {
         String cmdName = null;
-        if (this.getClass().getAnnotation(APICommand.class) != null) {
-            cmdName = this.getClass().getAnnotation(APICommand.class).name();
+        APICommand apiClassAnnotation = clazz.getAnnotation(APICommand.class);
+
+        if (apiClassAnnotation != null && apiClassAnnotation.name() != null) {
+            cmdName = apiClassAnnotation.name();
         } else {
-            cmdName = this.getClass().getName();
+            cmdName = clazz.getName();
```

Comment:
@ezntt 
can you please check if there is an api that the apiClassAnnotation is null or name is null ?

in other apis,  the command name is not class name.

Reply:
@weizhouapache 

No. as far as I've seen, there are no null annotations or null names throughout the APIs.
In my view, the main purpose of this PR is just to clean up the code and not change its behaviors if needed.

Reply:
@weizhouapache, I checked the code and did not find `APICommand` annotations with null name. Regarding the fallback to the class name, it is an already existing behavior, it was not introduced by this PR; therefore, if we had this situation, we probably would have catch it before.

Reply:
@ezntt thanks for checking. it looks good to me.

Reply:
@ezntt 
just a suggestion, maybe we can remove the `name` value from `APICommand` annotations in most apis.
As I commented before, the api name is same as
```
// TODO: remove "Cmd" and "CmdByAdmin" from class name.
```